module/zfs/zfs_ioctl.c

```diff
@@ -2415,6 +2450,11 @@ zfs_prop_set_special(const char *dsname, zprop_source_t source,
 		if (err == 0)
 			err = -1;
 		break;
+	case ZFS_PROP_KEYLOCATION:
+		err = dsl_crypto_can_set_keylocation(dsname, strval);
+		if (err == 0)
+			err = -1;
```

Comment:
I didn't do that because `ZFS_PROP_SNAPSHOT_LIMIT` explains that just a few lines above. I can add it though.